### PR TITLE
fix(ivy): ensure TestBed restores fields to the most original value

### PR DESCRIPTION
**Pull request by atscott**
If `TestBed.applyProviderOverrides` is called multiple times in a single test (happens if someone calls `compileModuleAsync` or `compileModuleSync`), the original fields for `ngComponentDef` are not restored correctly. Subsequent tests will get old versions of overridden providers in these cases.

### CODE REVIEW

File: packages/core/test/test_bed_spec.ts [hunk 1]
```diff
@@ -308,6 +308,42 @@ describe('TestBed', () => {
     expect(fixture.componentInstance.myProviders).toEqual([{value: 'new provider'}]);
   });
 
+  describe('allow to override provider with multiple calls to applyProviderOverrides', () => {
```
Comment (AndrewKushnir) [inline packages/core/test/test_bed_spec.ts]: The `applyProviderOverrides` function is not a part of TestBed's public API, so I think we can re-phrase title of this test suite (and comments in the test) to mention that the goal of this test is to make sure that there is no TestBed state leaking from one test to another.

Comment (atscott) [reply packages/core/test/test_bed_spec.ts]: Done.

**Review by AndrewKushnir — APPROVED** [review]
LGTM 👍

**Comment by atscott** [comment]
[Presubmit](https://test.corp.google.com/ui#id=OCL:271004462:BASE:271004485:1569365748196:1d66c520)
[Ivy Presubmit](https://test.corp.google.com/ui#id=OCL:271004462:BASE:271007430:1569366598648:424dcab6)

**Comment by atscott** [comment]
caretaker note: g3 failure is 92% flaky.

**Pull request closed without merging**